include/aos/common/tools/fs.hpp

```diff
@@ -358,8 +359,46 @@ class FS {
      */
     static Error ReadFileToString(const String& fileName, String& text)
     {
-        (void)fileName;
-        (void)text;
+        auto fd = open(fileName.CStr(), O_RDONLY);
+        if (fd != 0) {
+            return AOS_ERROR_WRAP(ErrorEnum::eFailed);
+        }
+
+        auto size = lseek(fd, 0, SEEK_END);
+        if (size < 0) {
+            close(fd);
+
+            return AOS_ERROR_WRAP(ErrorEnum::eFailed);
+        }
+
+        auto pos = lseek(fd, 0, SEEK_SET);
+        if (pos < 0) {
+            close(fd);
+
+            return AOS_ERROR_WRAP(ErrorEnum::eFailed);
+        }
+
+        auto err = text.Resize(size);
+        if (!err.IsNone()) {
+            close(fd);
+
+            return AOS_ERROR_WRAP(ErrorEnum::eFailed);
+        }
+
+        while (pos < size) {
```

Comment:
What if file will contain \0 in the middle?
What if it is binary file with random data?

Reply:
I think reading binary file in text mode is UB in general. This implementation follows the same idea.